### DISCUSSION TITLE
Connection Health Status?

### DISCUSSION THREAD

**Discussion by mbStavola** [nats-io/nats.rs#649] [Q&A] [answered]
As part of my service's health check I'd like to be able to inspect the health status of the NATS connection. I see that `Connection` will perform it's own PING/PONG health check in the background on some interval, but I don't see a way for me to get to that. Ideally I'd like to be able to issue my own PING check, but it's fine if I can just determine if the NATS connection is considered healthy at that point in time.

I'm using async-nats, by the way.

**Comment by Jarema** [comment, marked as answer]
You can use [even callback](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.event_callback) to get the info about state change.

```rust
async_nats::ConnectOptions::new().event_callback(|event| async move {
    match event {
    async_nats::Event::Disconnect => println!("disconnected"),
        async_nats::Event::Reconnect => println!("reconnected"),
        async_nats::Event::ClientError(err) => println!("client error occured: {}", err),
        other => println!("other event happened: {}", other),
}
}).connect("demo.nats.io").await?;
```



To get the current state, you can use
[connection state](https://docs.rs/async-nats/latest/async_nats/struct.Client.html#method.connection_state)

```rust
let client = async_nats::connect("demo.nats.io").await?;
println!("connection state: {}", client.connection_state());
```

You can also set the [ping interval](https://docs.rs/async-nats/latest/async_nats/struct.ConnectOptions.html#method.ping_interval) to increase the resolution at which connection state is checked when there is not a lot of activity / other signs of disconnect.

```rust
async_nats::ConnectOptions::new().flush_interval(Duration::from_secs(5)).connect("demo.nats.io").await?;
```

**Reply by mbStavola** [reply]
Ah! Thanks!

It seems I missed `connection_state()` as I am operating in a Jetstream context.
